app/routes/sprint-6/caseworkerManageRoutes.js

```diff
@@ -321,20 +321,46 @@ router.get("/referrals/:referralIndex/interventions/:interventionIndex/fast-forw
 });
 
 router.get("/referrals/:referralIndex/interventions/:interventionIndex/sessions/:sessionIndex/assessment", (req, res) => {
-    const intervention = findIntervention(req);
-    const referral = findReferral(req);
+    res.redirect(`/sprint-6/book-and-manage/manage-a-referral/caseworker/referrals/${req.params.referralIndex}/interventions/${req.params.interventionIndex}/sessions/${req.params.sessionIndex}/assessment/attended`);
+});
+
+for (const page of ["attended", "details"]) {
+    router.get(`/referrals/:referralIndex/interventions/:interventionIndex/sessions/:sessionIndex/assessment/${page}`, (req, res) => {
+	const intervention = findIntervention(req);
+	const referral = findReferral(req);
+
+	const sessionIndex = parseInt(req.params.sessionIndex);
+
+	res.render(`sprint-6/book-and-manage/manage-a-referral/caseworker/assessment-${page}`, { referralIndex: req.params.referralIndex, interventionIndex: req.params.interventionIndex, intervention, sessionIndex, referral });
+    });
+}
 
+function updateWipAssessment(req) {
+    const intervention = findIntervention(req);
     const sessionIndex = parseInt(req.params.sessionIndex);
+    const session = intervention.sessions[sessionIndex];
+
+    if (session.wipAssessment == null) {
```

Comment:
A couple of things here:
1. `session.wipAssessment` will be `undefined` rather than `null` if it's not been set on the session. This works because you're using the `==` operator which in this case converts the parameters to booleans (I think), so it's comparing it to a falsy value. Relying on `==` can get you into trouble, so it's recommended to use `===` in Javascript as it doesn't convert the parameters. [0]
2. Not a blocker, but I'd probably expect an `updateWipAssessment` function to take the assessment as an argument, which would probably mean doing this check outside this function and passing in an empty `wipAssessment`.

[0] https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons

Reply:
Thanks! That's good to know about the `== null` thing - I've been relying on this behaviour throughout the prototype and had a vague sense there was a better way to do it. I'll update it here, but I think that if you were to run a linter on my prototype code it would probably run away and hide in a cave for the next 60 years.

And I've changed the function name to `updateWipAssessmentFromRequest`.